gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -205,23 +208,23 @@ public WriteObjectResponse call() throws IOException {
       }
     }
 
-    class OutOfBufferedDataException extends IOException {
-      public OutOfBufferedDataException(String message) {
-        super(message);
-      }
-    }
-
     boolean isRetriableError(Throwable throwable) {
-      if (throwable instanceof OutOfBufferedDataException) return false;
-      Throwable cause = throwable.getCause();
-      if (cause == null) return true;
-      return isRetriableError(cause);
+      Status status = Status.fromThrowable(throwable);
+      Status.Code statusCode = status.getCode();
+      return (TRANSIENT_ERRORS.contains(statusCode));
     }
 
     private WriteObjectResponse doResumableUpload() throws IOException {
+      long committedWriteOffset = 0;
```

Comment:
Why do we need both `writeOffset` and `committedWriteOffset` ? We should only rely on the offset from the server

Reply:
`writeOffset` tracks the client offset, it is used as we are writing the chunks. `committedWriteOffset` tracks server side offset and is used to clear the local buffers. Hence we need both of them.